src/likelihoods/gaussian.jl

```diff
@@ -10,13 +10,15 @@ uncertainity associated with the data follows a Gaussian distribution.
 On calling, this would return a normal distribution with mean `f` and variance σ².
 """
 struct GaussianLikelihood{T<:Real}
-    σ²::T
+    σ²::Vector{T}
```

Comment:
Do we need things to be in a vector form if we are using `Functors`?

Reply:
Also, shouldn't we have a link here too? We should be able to use Links regardless of whether the parameters of a distribution are constrained.

Reply:
`IdentityLink` could be the default.

Reply:
Here a link is not needed because `f` is just used as the mean of the Gaussian and is already in the right domain

Reply:
The Vector form is to insure compatibility with Zygote

Reply:
> Here a link is not needed because f is just used as the mean of the Gaussian and is already in the right domain

Yes but I just wanted to give that flexibility. I think `LinkedLikelihood` will serve that purpose.